File: src/org/genyris/exception/GenyrisUserException.java
package org.genyris.exception;

import org.genyris.core.Exp;
import org.genyris.interp.Environment;


public class GenyrisUserException extends GenyrisException {

    private static final long serialVersionUID = -2590161545543312593L;

    public GenyrisUserException(Exp exp, Environment env) {
        super(exp);
    }
	public String getMessage() {
		return GenyrisUserException.class.toString() + ": " 
		+ (_reason != null ? _reason.toString() : super.getMessage());
	}	
}

File: src/org/genyris/exception/GenyrisUserException.kt
package org.genyris.exception

import org.genyris.core.Exp
import org.genyris.interp.Environment

class GenyrisUserException(exp: Exp, env: Environment?) : GenyrisException(exp) {
    companion object {
        private val serialVersionUID = -2590161545543312593L
    }
}
